### PR TITLE
fix(ai): update useAIGeneration to add graphqlErrors before return

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Danny noticed an error where simple string type returns from the client were returning malformed objects
- This is because we were assuming all return types from the generate client would be spreadable objects
- Updated called to useDataState to just return the data and massage the return type to add graphqlErrors after to avoid having to spread the return data

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]

**Review by calebpollman — COMMENTED** [review]
The issue here reads as misuse of `useDataState` and not necessarily an issue with `useDataState` itself. Adding `graphqlErrors` to the `DataState` output itself can be done by massaging the output of `useDataState` before returning from `useAIGeneration`

**Comment by thaddmt** [comment]
> The issue here reads as misuse of `useDataState` and not necessarily an issue with `useDataState` itself. Adding `graphqlErrors` to the `DataState` output itself can be done by massaging the output of `useDataState` before returning from `useAIGeneration`

Ended up going with your solution, great call out, thanks for catching

**Review by dbanksdesign — DISMISSED** [review]

File: packages/react-ai/src/hooks/useAIGeneration.tsx [hunk 1]
```diff
@@ -45,31 +55,37 @@ export function createUseAIGeneration<
...
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return data;
     };
 
-    return useDataState(updateAIGenerationStateAction, {});
+    const [data, handler] = useDataState(
+      updateAIGenerationStateAction,
+      undefined
+    );
+    return [{ ...data, graphqlErrors }, handler];
```
Comment (calebpollman) [inline packages/react-ai/src/hooks/useAIGeneration.tsx:88]: Can simplify a bit and remove the `graphqlErrors` state by returning `result` directly from `updateAIGenerationStateAction` and restructuring the hook return output, something like:
```ts
    const { data, errors } =
      (result?.data as SingularReturnValue<Schema[Key]['returnType']>) ?? {};

    const output: [
      state: GenerateState<Schema[Key]['returnType']>,
      handleAction: (input: Schema[Key]['args']) => void,
    ] = [{ ...result, data, graphqlErrors: errors }, handler];
```

Comment (thaddmt) [reply packages/react-ai/src/hooks/useAIGeneration.tsx:88]: Ah damn this is much cleaner, let me try that and ping you :)

**Review by dbanksdesign — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**